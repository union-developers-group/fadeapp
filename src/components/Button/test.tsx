import { render, screen } from '@testing-library/react'

import { Button } from '.'

const { getByRole } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    render(<Button>Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('py-4 w-[21.5rem] text-lg h-[3.625rem]')
  })

  it('should render with custom class', () => {
    render(<Button className="uppercase">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass(
      'py-4 w-[21.5rem] text-lg h-[3.625rem] uppercase',
    )
  })

  it('should render with correct sizes', () => {
    const { rerender } = render(<Button size="small">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('w-[16.188rem] text-2xl h-[2.875rem]')

    rerender(<Button size="medium">Children</Button>)

    expect(component).toHaveClass('py-4 w-[21.5rem] text-lg h-[3.625rem]')

    rerender(<Button size="large">Children</Button>)

    expect(component).toHaveClass('py-3 w-[22.188rem] text-lg h-[2.813rem]')
  })

  it('should match snapshot', () => {
    const { container } = render(<Button>Children</Button>)

    expect(container).toMatchSnapshot()
  })
})
