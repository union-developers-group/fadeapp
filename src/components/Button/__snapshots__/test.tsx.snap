// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="flex items-center justify-center rounded-10 bg-primary font-bold text-secondary transition hover:text-primary hover:bg-transparent border-2 border-primary transition-color py-4 w-[21.5rem] text-lg h-[3.625rem] "
  >
    Children
  </button>
</div>
`;
